addons/web_editor/models/ir_ui_view.py

```diff
@@ -134,19 +130,25 @@ def _copy_field_terms_translations(self, record_from, name_field_from, record_to
         if not field_to.store:
             raise ValueError("Field %r is not stored" % field_to)
 
+        # This will also implicitly check for `read` access rights
+        if not record_to[name_field_to] or not any(recordset_from.mapped(name_field_from)):
+            return
+
         lang_env = self.env.lang or 'en_US'
         langs = set(lang for lang, _ in self.env['res.lang'].get_installed())
 
         # 1. Get translations
-        record_from.flush_model([name_field_from])
+        recordset_from.flush_model([name_field_from])
         existing_translation_dictionary = field_to.get_translation_dictionary(
             record_to[name_field_to],
             {lang: record_to.with_context(prefetch_langs=True, lang=lang)[name_field_to] for lang in langs if lang != lang_env}
         )
-        extra_translation_dictionary = field_from.get_translation_dictionary(
-            record_from[name_field_from],
-            {lang: record_from.with_context(prefetch_langs=True, lang=lang)[name_field_from] for lang in langs if lang != lang_env}
-        )
+        extra_translation_dictionary = {}
+        for rec_from in recordset_from:
```

Comment:
Nitpicking/ignore if you pref: naming it `record_from` will make it so there is no diff (in ignore whitespace) in the loop code lines:
```suggestion
        for record_from in recordset_from:
```